core/trino-main/src/test/java/io/trino/sql/TestExpressionInterpreter.java

```diff
@@ -473,6 +473,11 @@ public void testIn()
 
         assertOptimizedEquals("NULL IN (2, NULL, 3, 5)", "NULL");
         assertOptimizedEquals("3 IN (2, NULL)", "NULL");
+        assertOptimizedEquals("3 IN (NULL)", "NULL");
+        assertOptimizedEquals("'3' IN (CAST(2 AS varchar), NULL)", "NULL");
+        assertOptimizedEquals("3 IN (CAST(2 AS bigint), CAST(CAST(NULL AS varchar) AS bigint) + NULL) ", "NULL");
```

Comment:
The tests added here seem to pass without the main change. Let's move them to a separate commit.